docs/qppe.yaml

```diff
@@ -806,3 +825,148 @@ components:
           description: Human-readable reason why there was an error.
       required: [ code ]
       additionalProperties: false
+
+    FormLabel:
+      type: string
+      description: The displayed label for this form element
+    FormName:
+      type: string
+      pattern: ^[a-zA-Z_][a-zA-Z0-9_]*$
+      description: The key of the form data field produced by this form element
+
+    TextElement:
+      type: object
+      properties:
+        kind:
+          type: string
+          enum: [ text ]
+        label:
+          $ref: "#/components/schemas/FormLabel"
+        text:
+          type: string
+          description: Static text to display
+      required: [ kind, label, text ]
+
+    TextInputElement:
+      type: object
+      properties:
+        kind:
+          type: string
+          enum: [ input ]
+        label:
+          $ref: "#/components/schemas/FormLabel"
+        name:
+          $ref: "#/components/schemas/FormName"
+        required:
+          type: boolean
+          default: false
+      required: [ kind, label, name ]
+
+    CheckboxElement:
+      type: object
+      properties:
+        kind:
+          type: string
+          enum: [ checkbox ]
+        label:
+          $ref: "#/components/schemas/FormLabel"
+        name:
+          $ref: "#/components/schemas/FormName"
+        required:
+          type: boolean
+          default: false
+      required: [ kind, label, name ]
+
+    CheckboxGroupElement:
```

Comment:
Grundsätzlich würde ich das Konzept, mehrere Elemente gruppieren zu können, auf beliebige Elemente erweitern, eben wie bei Moodle.

Reply:
Wie [`addGroup`](https://docs.moodle.org/dev/lib/formslib.php_Form_Definition#addGroup) meinst du, dass die gruppierten Elemente auf einer Zeile mit einem gemeinsamen label angezeigt werden? Ja, ist vermutlich sinnvoll. Das hier soll aber das pendant zum [checkbox controller](https://moodledev.io/docs/apis/subsystems/form/advanced/checkbox-controller) sein.

Reply:
Habe jetzt noch ein GroupElement hinzugefügt. Ich bin mir noch unsicher, wie das Zusammenspiel mit dem CheckboxGroupElement laufen wird. Möglicherweise müssen ich das dann wieder streichen und wie in Moodle mit einem separaten CheckboxControllerElement abbilden :thinking:

Reply:
Ja, das sehen wir dann später, ob ggf. wieder Änderungen notwendig sind. Soweit erscheint mir das stimmig.